### ISSUE TITLE
type alias `Plus2` is never used

### ISSUE THREAD

**Issue by IceTDrinker** [rust-lang/rust-clippy#13121]
### Summary

Some type is flagged by Clippy as never used, though it is used in an internal trait implementation

### Lint Name

dead_code

### Reproducer

I tried this code:

clone https://github.com/zama-ai/concrete-fft/commit/aeb75d4b7c5cc6b3031bf5b9f4730961f3132fe5

make pcc (pre commit checks, runs a clippy check)

faulty clippy invokation:
```
RUSTFLAGS="-C target-cpu=native" cargo "+nightly-2024-07-18" clippy --all-targets \
        --features=serde -- --no-deps -D warnings
```

I saw this happen:

```
error: type alias `Plus2` is never used
  --> src/nat.rs:26:10
   |
26 | pub type Plus2<N> = Successor<Plus1<N>>;
   |          ^^^^^
   |
   = note: `-D dead-code` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(dead_code)]`

error: type alias `Plus3` is never used
  --> src/nat.rs:27:10
   |
27 | pub type Plus3<N> = Successor<Plus2<N>>;
   |          ^^^^^

error: type alias `Plus4` is never used
  --> src/nat.rs:28:10
   |
28 | pub type Plus4<N> = Successor<Plus3<N>>;
   |          ^^^^^

error: could not compile `concrete-fft` (lib) due to 3 previous errors
```

I expected to see this happen:

No warning/error for these types as they are used in e.g. https://github.com/zama-ai/concrete-fft/blob/aeb75d4b7c5cc6b3031bf5b9f4730961f3132fe5/src/dit16.rs#L823

### Version

```text
rustc 1.81.0-nightly (fcc325f1b 2024-07-17)
binary: rustc
commit-hash: fcc325f1bc477975e2ce5ba534fe4c77ff8a8536
commit-date: 2024-07-17
host: x86_64-unknown-linux-gnu
release: 1.81.0-nightly
LLVM version: 18.1.7
```

**Comment by Alexendoo** [comment]
`dead_code` comes from rustc, does `cargo +nightly-2024-07-18 check --all-targets --features serde` exhibit the same behaviour?

**Comment by IceTDrinker** [comment]
checking

**Comment by IceTDrinker** [comment]
warnings are popping up as well, so wrong repo I guess ? is there a way to transfer this issue to rustc ?
